dlib/test/math.cpp

```diff
@@ -0,0 +1,497 @@
+            0.36907445, 0.39697806, 0.42547675, 0.45447258, 0.48386148,
+            0.51353385, 0.54337511, 0.57326642, 0.60308532, 0.63270649,
+            0.66200256, 0.69084482, 0.71910415, 0.74665178, 0.77336018,
+            0.7991039 , 0.82376041, 0.84721094, 0.86934129, 0.89004263,
+            0.90921226, 0.92675431, 0.94258044, 0.95661046, 0.96877289,
+            0.97900548, 0.98725567, 0.99348091, 0.99764907, 0.99973856,
+            0.99973856, 0.99764907, 0.99348091, 0.98725567, 0.97900548,
+            0.96877289, 0.95661046, 0.94258044, 0.92675431, 0.90921226,
+            0.89004263, 0.86934129, 0.84721094, 0.82376041, 0.7991039 ,
+            0.77336018, 0.74665178, 0.71910415, 0.69084482, 0.66200256,
+            0.63270649, 0.60308532, 0.57326642, 0.54337511, 0.51353385,
+            0.48386148, 0.45447258, 0.42547675, 0.39697806, 0.36907445,
+            0.34185729, 0.3154109 , 0.28981222, 0.26513048, 0.24142695,
+            0.21875482, 0.19715903, 0.1766763 , 0.15733509, 0.13915573,
+            0.12215057, 0.1063242 , 0.09167367, 0.07818886, 0.06585282,
+            0.05464219, 0.04452765, 0.03547437, 0.02744256, 0.020388
+    };
+
+    /*! Data generated using scipy.signal.windows.kaiser(100, beta=(0.1102*(60.0 - 8.7)), sym=False) !*/
+    const static double KAISER_DATA_N_100_ATT_60_PERIODIC[100] = {
+            0.020388  , 0.02736726, 0.03530386, 0.0442408 , 0.05421678,
+            0.06526578, 0.0774165 , 0.09069194, 0.10510897, 0.12067792,
+            0.13740224, 0.15527819, 0.17429461, 0.19443265, 0.21566568,
+            0.23795919, 0.2612707 , 0.28554985, 0.31073847, 0.33677071,
+            0.3635733 , 0.39106578, 0.41916087, 0.44776486, 0.47677808,
+            0.50609537, 0.5356067 , 0.56519774, 0.59475052, 0.62414415,
+            0.65325551, 0.68196003, 0.71013245, 0.73764762, 0.76438133,
+            0.79021107, 0.81501689, 0.83868214, 0.86109433, 0.88214579,
+            0.90173451, 0.91976477, 0.9361478 , 0.95080246, 0.96365572,
+            0.97464323, 0.98370975, 0.99080954, 0.99590669, 0.99897537,
+            1.        , 0.99897537, 0.99590669, 0.99080954, 0.98370975,
+            0.97464323, 0.96365572, 0.95080246, 0.9361478 , 0.91976477,
+            0.90173451, 0.88214579, 0.86109433, 0.83868214, 0.81501689,
+            0.79021107, 0.76438133, 0.73764762, 0.71013245, 0.68196003,
+            0.65325551, 0.62414415, 0.59475052, 0.56519774, 0.5356067 ,
+            0.50609537, 0.47677808, 0.44776486, 0.41916087, 0.39106578,
+            0.3635733 , 0.33677071, 0.31073847, 0.28554985, 0.2612707 ,
+            0.23795919, 0.21566568, 0.19443265, 0.17429461, 0.15527819,
+            0.13740224, 0.12067792, 0.10510897, 0.09069194, 0.0774165 ,
+            0.06526578, 0.05421678, 0.0442408 , 0.03530386, 0.02736726
+    };
+
+    template<typename R>
+    void test_kaiser()
+    {
+        constexpr R tol = std::is_same<R,float>::value ? 1e-3 : 1e-7;
```

Comment:
Can these tolerances be way lower too?  Presumably saving out the test data with more digits is all that's needed?

Reply:
Yes I can do that

Reply:
Can't for kaiser as the bessel function implementation is a bit naive and therefore less accurate than boost. The boost implementation is unbelievably complicated and unreadable. dlib's is much more explicit and readable. That comes at a cost though.